tests/native/test_resume.py

```diff
@@ -0,0 +1,62 @@
+import unittest
+from manticore.native import Manticore
+from manticore.core.plugin import Plugin
+from manticore.core.state import SerializeState, TerminateState
+from pathlib import Path
+
+
+class InsnCounterPlugin(Plugin):
+    def did_execute_instruction_callback(self, state, last_pc, pc, instruction):
+        with self.locked_context("counter", dict) as ctx:
+            val = ctx.setdefault(instruction.mnemonic, 0)
+            ctx[instruction.mnemonic] = val + 1
```

Comment:
I'm curious about the `with self.locked_context`.  This seems to be a context shared by all states, correct?

When saving and loading a single state, there could be other states generated by forking with program exploration, so I think these counts may be off in the tests after saving and loading only one state and comparing it to a full run (`counts_canonical`). Would keying the counts with `state.id` or something be better?

Reply:
The instruction I put the hook at should be before any forking takes place, and the total counts across all states should be consistent. At least, I think they should. This test works just fine on OSX (cheating, I know) but fails on linux, with most of the instruction counts being slightly off. They're not _gone_ in either case, so it doesn't seem like a multiprocessing issue as I had initially expected. Perhaps we are missing a state somewhere? I'll keep looking into it...

Reply:
> The instruction I put the hook at should be before any forking takes place

Maybe a comment in the code about this at the hooked address would help, but this makes sense otherwise.

Reply:
Hmm, for some reason, instruction counts on Linux are not consistent across runs. That would explain why the test is failing. Perhaps I need a better metric for measuring native success.

Reply:
Fixed, kinda. Now I check the state counts at different points in execution. Not quite as precise as the WASM tests, but close enough.